backend/src/database/repositories/memberRepository.ts

```diff
@@ -3694,6 +3694,67 @@ class MemberRepository {
       ids: members.map((i: any) => i.id),
     }
   }
+
+  static async moveNotesBetweenMembers(
+    fromMemberId: string,
+    toMemberId: string,
+    options: IRepositoryOptions,
+  ): Promise<void> {
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const seq = SequelizeRepository.getSequelize(options)
+
+    const query = `
+      update "memberNotes" set "memberId" = :toMemberId where "memberId" = :fromMemberId;
+    `
+
+    await seq.query(query, {
+      replacements: {
+        fromMemberId,
+        toMemberId,
+      },
+      type: QueryTypes.UPDATE,
+      transaction,
+    })
+  }
+
+  static async moveTasksBetweenMembers(
+    fromMemberId: string,
+    toMemberId: string,
+    options: IRepositoryOptions,
+  ): Promise<void> {
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const seq = SequelizeRepository.getSequelize(options)
+
+    const params: any = {
+      fromMemberId,
+      toMemberId,
+    }
+
+    const deleteQuery = `
+      delete from "memberTasks" using "memberTasks" as mt2
+      where "memberTasks"."memberId" = :fromMemberId 
+      and "memberTasks"."taskId" = mt2."taskId"
+      and mt2."memberId" = :toMemberId;
```

Comment:
Is this because a task can belong to both members? If yes, why don't we do the same thing with notes - I think they're similar in structure, aren't they?

Reply:
No, the structure is a bit different. Notes are one-to-one, and tasks are one-to-many in structure. So, when I was testing, there was a case where one task was associated with two members. Merging them caused duplicate errors when we updated the `memberId` related to tasks.

Reply:
I think both are many to many
https://github.com/CrowdDotDev/crowd.dev/blob/f72276de363ca5724029ab4a7504e99ee06a6182/backend/src/database/models/member.ts#L124-L127
https://github.com/CrowdDotDev/crowd.dev/blob/f72276de363ca5724029ab4a7504e99ee06a6182/backend/src/database/models/note.ts#L32-L36

but it's not possible to assign the same note to multiple members through the UI so it's not a case that can happen frequently - Let's add it anyways to be complete what do u think? 
Can you double check if same note can belong to two different members in db layer?

Reply:
Good catch, you're right; Indeed, they have the same structure in the database and function similar to tasks but for some reason, we do not support that in the UI. I've added the code for notes to handle the case and also the tests for the same to make it complete. Thanks! 😄